File: load-fonts_webgl10-typescript/public/js/Scene.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Textures_1 = require("./Engine/Resources/Textures");
var EngineCore_1 = require("./Engine/EngineCore");
var Camera_1 = require("./Engine/Camera");
var gl_matrix_1 = require("gl-matrix");
var FontRenderable_1 = require("./Engine/Renderables/FontRenderable");
var Screen = /** @class */ (function () {
    function Screen() {
    }
    Screen.prototype.Load = function () {
        var textureName = "assets/fonts/system-default-font.png";
        Textures_1.default.Instance.LoadTexture(textureName);
        EngineCore_1.default.Instance.ClearCanvas(gl_matrix_1.vec4.fromValues(0.282, 0.498, 0.282, 1.0));
        var gl = EngineCore_1.default.Instance.gl;
        var camera = new Camera_1.default(gl_matrix_1.vec3.fromValues(0, 0, 0), 10, gl_matrix_1.vec4.fromValues(20, 20, gl.canvas.width - 40, gl.canvas.height - 40));
        camera.BackgroundColor = gl_matrix_1.vec4.fromValues(0.905, 0.850, 0.752, 1);
        camera.SetupViewProjection();
        var textSystemFont = new FontRenderable_1.default("Hello, World!");
        this.InitText(textSystemFont, -3, 0, gl_matrix_1.vec4.fromValues(0.282, 0.443, 0.498, 1), 1);
        textSystemFont.Draw(camera.VPMatrix);
        // let square = new Renderable();
        // square.Draw(camera.VPMatrix);
    };
    Screen.prototype.InitText = function (font, posX, posY, color, textHeight) {
        font.Color = color;
        font.XForm.SetPosition(posX, posY);
        font.Height = textHeight;
    };
    return Screen;
}());
exports.default = Screen;

File: LoadFonts_OpenTkOpenGL31CSharp/LoadFonts/Scene.cs

using Engine;
using OpenTK;
using OpenTK.Graphics;

class Scene
{
    private Color4 _bgColor;
    private Camera _camera;
    private int _windowWidth;
    private int _windowHeight;
    private FontRenderable _textSystemFont;

    public bool IsReady { get; set; }

    public Scene(int windowWidth, int windowHeight)
    {
        _bgColor = new Color4(0.282f, 0.498f, 0.282f, 1f);
        _windowWidth = windowWidth;
        _windowHeight = windowHeight;

        IsReady = false;
    }

    public void Load()
    {
        if (!IsReady) return;
    }

    public void Initialize()
    {
        IsReady = true;

        _camera = new Camera(
            wcCenter: Vector2.Zero,
            wcWidth: 10,
            viewport: new Vector4(20, 20, _windowWidth - 40, _windowHeight - 40));

        _camera.BackgroundColor = new Color4(0.905f, 0.850f, 0.752f, 1f);

        _textSystemFont = new FontRenderable("Hello, World!");
        InitText(_textSystemFont, -3, 0, new Color4(0.282f, 0.443f, 0.498f, 1f), 1f);
    }

    public void Draw()
    {
        if (!IsReady) return;

        EngineCore.Instance.ClearCanvas(_bgColor);
        _camera.SetupViewProjection();
        _textSystemFont.Draw(_camera.VPMatrix);
    }

    private void InitText(FontRenderable font, float posX, float posY, Color4 color, float textHeight)
    {
        font.XForm.SetPosition(posX, posY);
        font.Color = color;
        font.Height = textHeight;
    }
}
